### PR TITLE
eq-1246 - Refactor Assertion Tests

### PR DESCRIPTION
**Pull request by MIslam92**
Context of this PR:
Refactoring of tests to remove all assertions in code and to be replaced with exceptions & also to add tests for these.
Also, new methods added to deal with "Location" in the answer_store

Trello Card: https://trello.com/c/93EgMm8q

How to review:
Run tests

### CODE REVIEW

**Review by shahi645 — COMMENTED** [review]
LGTM 👍🏿

**Review by shahi645 — APPROVED** [review]

File: app/submitter/converter.py [hunk 1]
```diff
@@ -24,6 +24,7 @@ def convert_answers(metadata, questionnaire_json, answer_store, routing_path, fl
     :param questionnaire_json: the questionnaire json
     :param answer_store: the users answers
     :param routing_path: the path followed by the user when answering the questionnaire
+    :param flushed: writes file data to disk
```
Comment (jonnyshaw89) [inline app/submitter/converter.py]: Does it?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/submitter/converter.py (hunk 1 above)
Comment (MIslam92) [inline app/submitter/converter.py]: ¯\_(ツ)_/¯

File: app/forms/household_composition_form.py [hunk 2]
```diff
@@ -7,6 +9,8 @@
 
 from werkzeug.datastructures import MultiDict
 
+logger = wrap_logger(logging.getLogger(__name__))
```
Comment (jonnyshaw89) [inline app/forms/household_composition_form.py]: we should stick to the standard of
```
from structlog import get_logger
logger = get_logger()
```

File: app/helpers/schema_helper.py [hunk 3]
```diff
@@ -127,7 +127,8 @@ def get_aliases(cls, survey_json):
             for question in cls.get_questions_for_block(block):
                 for answer in question['answers']:
                     if 'alias' in answer:
-                        assert answer['alias'] not in aliases, 'Duplicate alias found: ' + answer['alias']
+                        if answer['alias'] in aliases:
+                            raise AssertionError('Duplicate alias found: ' + answer['alias'])
```
Comment (jonnyshaw89) [inline app/helpers/schema_helper.py]: Do we still want to throw this type of Error?

File: tests/app/helpers/test_schema_helper.py [hunk 4]
```diff
@@ -97,6 +97,33 @@ def test_get_parent_options_for_block(self):
...
+            "groups": [{
+                "blocks": [{
+                    "sections": [{
+                        "questions": [{
+                            "answers": [{
+                                "id": "1",
+                                "alias": "alias_name",
+                                "type": "Checkbox"
+                                },
+                                        {
```
Comment (jonnyshaw89) [inline tests/app/helpers/test_schema_helper.py]: Incorrect indentation

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/questionnaire/rules.py [hunk 5]
```diff
@@ -114,7 +114,9 @@ def evaluate_when_rules(when_rules, metadata, answer_store, group_instance):
             answer_index = when_rule['id']
             filtered = answer_store.filter(answer_id=answer_index, group_instance=group_instance)
 
-            assert len(filtered) <= 1, "Condition will not met: Multiple ({:d}) answers found".format(len(filtered))
+            if len(filtered) > 1:
+                raise AssertionError("Multiple answers ({:d}) found evaluating when rule for answer ({})"
```
Comment (jonnyshaw89) [inline app/questionnaire/rules.py]: There was more than just the 1 AssertionError

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

**Review by jonnyshaw89 — APPROVED** [review]

**Comment by jonnyshaw89** [comment]
@MIslam92 Can you squash this branch?

**Comment by shahi645** [comment]
@jonnyshaw89 He is away until Tuesday, want me to do it ?

**Pull request merged**